File: 2490.回环句.go
/*
 * @lc app=leetcode.cn id=2490 lang=golang
 *
 * [2490] 回环句
 */

// @lc code=start

func isCircularSentence(sentence string) bool {
	for i := 0; i < len(sentence); i++ {
		if sentence[i] == ' ' {
			if sentence[i-1] != sentence[i+1] {
				return false
			}
		}
	}
	return sentence[0] == sentence[len(sentence)-1]
}

// @lc code=end

File: 2490.回环句.c
/*
 * @lc app=leetcode.cn id=2490 lang=c
 *
 * [2490] 回环句
 */

 // @lc code=start
bool isCircularSentence(char* sentence) {
    int n = strlen(sentence);
    for (int i = 0; i < n; i++) {
        if (sentence[i] == ' ') {
            if (sentence[i - 1] != sentence[i + 1]) {
                return false;
            }
        }
    }
    return sentence[0] == sentence[n - 1];
}
// @lc code=end
